pkg/storage/client_raft_test.go

```diff
@@ -1894,6 +1894,84 @@ func TestQuotaPool(t *testing.T) {
 	}
 }
 
+// TestWedgedReplicaDetection verifies that a leader replica is able to correctly detect a wedged follower replica.
+func TestWedgedReplicaDetection(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+
+	const numReplicas = 3
+	const rangeID = 1
+
+	var manuals []*hlc.ManualClock
+	var clocks []*hlc.Clock
+	for i := 0; i < numReplicas; i++ {
+		manuals = append(manuals, hlc.NewManualClock(123))
+		clocks = append(clocks, hlc.NewClock(manuals[i].UnixNano, time.Nanosecond))
+	}
+
+	sc := storage.TestStoreConfig(nil)
+	// Suppress timeout-based elections to avoid leadership changes in ways
+	// this test doesn't expect.
+	sc.RaftElectionTimeoutTicks = 100000
+	mtc := &multiTestContext{storeConfig: &sc, clocks: clocks}
+	mtc.Start(t, numReplicas)
+	defer mtc.Stop()
+
+	mtc.replicateRange(rangeID, 1, 2)
+
+	leaderRepl := mtc.getRaftLeader(rangeID)
+	// We need four steps in this test:
```

Comment:
Thanks for your comments!

Forgive my poor English expression. @nvanbenschoten